tests/tests/worm-xattr-unit.sh

```diff
@@ -0,0 +1,49 @@
+t_require_commands touch rm setfattr
+
+echo "== test creation"
+touch "$T_D0/file-1"
+SECS=$(date '+%s')
+NSECS=$(date '+%N')
+DELAY=10
+EXP=$(( $SECS + $DELAY ))
+setfattr -n scoutfs.worm.v1_expiration -v $EXP.$NSECS "$T_D0/file-1"
+
+echo "== test value is set correctly"
+getfattr --absolute-names -d -m - "$T_D0/file-1" | grep $EXP.$NSECS &> /dev/null
+if [ $? == 0 ]; then
+    echo "== getfattr matches setfattr"
+fi
+
+echo "== test deletion of xattr 1 sec after setting"
+sleep 1
+setfattr -x scoutfs.worm.v1_expiration "$T_D0/file-1"
+
+echo "== test setting xattr to a previous time after setting"
+setfattr -n scoutfs.worm.v1_expiration -v $SECS.$NSECS "$T_D0/file-1"
+
+echo "== test deletion before expiration"
+rm -f "$T_D0/file-1"
+
+echo "== Try to move the file prior to expiration"
+mv $T_D0/file-1 $T_D0/file-2
+
+echo "== Try to write to file before expiration"
+date >> $T_D0/file-1
+
+echo "== wait til expiration"
+sleep $DELAY
```

Comment:
Maybe add a second or two just to proactively avoid spurious failures from crummy guests that get stuck in IO for a bit?   There's already that baked in second delay from the initial sleep before deletion so it's probably fine, but a bit more headroom wouldn't hurt.  (fancier still would be a loop sleeping from the current time 'till the delay and breaking once we're past, but meh.)